### ISSUE TITLE
Add user and user group permissions support

### ISSUE THREAD

**Issue by reinoudk** [rewe-digital/terraform-provider-sonarcloud#81]
### Description

The provider used to have a `sonarcloud_user_group_permissions` resource, which was removed due to problems with the API. 
Testing locally with the `api/permissions/users` and `api/permissions/groups` has worked well, so I think the `user_group_permissions` resource can be reintroduced, along with the new `user_permissions` resource. 

#### Data sources 

* sonarcloud_user_groups_permissions
* sonarcloud_users_permissions

#### Resources

* sonarcloud_user_group_permissions
* sonarcloud_user_permissions

#### Data sources

```tf
# Global users permissions
data "sonarcloud_users_permissions" "global_all" {}

# Users permissions on a specific object
data "sonarcloud_users_permissions" "project_all" {
  project = "my-project"
}
```

```tf
# Global groups permissions
data "sonarcloud_user_groups_permissions" "global_all" {}

# Groups permissions on a specific object
data "sonarcloud_user_group_permissions" "project_all" {
  project = "my-project"
}
```

#### Resources

```tf
# Global user permissions
resource "sonarcloud_user_permissions" "global" {
  user = "some-user"
  permissions = ["scan"]
}

# User permissions on a specific object
resource "sonarcloud_user_permissions" "project" {
  project = "my-project"
  user = "some-user"
  permissions = ["admin", "scan"]
}
```

```tf
# Global group permissions
resource "sonarcloud_user_group_permissions" "global" {
  name = "my-group"
  permissions = ["scan"]
}

# Group permissions on a specific object
resource "sonarcloud_user_group_permissions" "project" {
  project = "my-project"
  name = "my-group"
  permissions = ["admin", "scan"]
}
```

### References

The new API endpoints are not documented yet. There's an [official post](https://community.sonarsource.com/t/how-to-reverse-engineer-the-sonar-web-api/66810) about reverse engineering the API, however, so this seems to be "encouraged". 

Support for the endpoints will have to be added to [go-sonarcloud](https://github.com/reinoudk/go-sonarcloud) though.

**Comment by severin** [comment]
I would love to have this available, it's the only missing piece to completely describe our Sonarcloud setup in TF.

**Comment by reinoudk** [comment]
Hi @severin, 
I've had some time this week to look at this and the tests are looking green for the `sonarcloud_user_group_permissions` resource. I'm going to do some refactoring in `go-sonarcloud` to make the generic API calls a bit more straightforward to use, but this shouldn't take too long. Together with implementing the `sonarcloud_user_permissions` resource and the two data sources I'd recon I should be able to have this done somewhere next week.

**Comment by severin** [comment]
Wow, great news!

**Comment by reinoudk** [comment]
The resources and data sources have been added and are available in `v0.5.0` of the provider 🙂 .
